GoogleSignIn/Sources/Public/GoogleSignIn/GIDSignIn.h

```diff
@@ -66,6 +67,22 @@ typedef NS_ERROR_ENUM(kGIDSignInErrorDomain, GIDSignInErrorCode) {
 /// The active configuration for this instance of `GIDSignIn`.
 @property(nonatomic, nullable) GIDConfiguration *configuration;
 
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+/// Configures `GIDSignIn` for use.
+///
+/// @param completion A nullable callback block passing back any error arising from the
+/// configuration process if any exists.
+///
+/// Call this method on `GIDSignIn` prior to use and as early as possible. This method generates App
+/// Attest key IDs and the attestation object eagerly to minimize latency later on during the sign
+/// in or add scopes flows.
```

Comment:
With regards to the potential error with code `kGIDAppCheckAlreadyPrepared` here, I think there are two scenarios:
1. The developer has called `configureWithCompletion:` twice in their code and the `kGIDAppCheckAlreadyPrepared` error is actionable (i.e., remove the extraneous configure call).
2. The app is being re-launched and is already prepared. Since developers are always (AFAICT) expected to call this method, I _think_ the completion block being called with an `kGIDAppCheckAlreadyPrepared` error may still be confusing (since they'd need to dig through the source code to see the comment that it's not a failing error). In this case the error isn't actionable and they should just ignore it.

Suggestion: IMHO, I think it would be ideal if the completion block was called with an error in scenario 1 but not scenario 2, potentially doing something like:
```
#if DEBUG
  NSLog(@"<INFO> App Check already prepared.");
#endif
```
Or not logging / emitting an error at all. Another alternative would be to always log a notice. I just worry that an `NSError` will be interpreted as an error, not a notice / info, regardless of the documentation around the `kGIDAppCheckAlreadyPrepared` error code.

Note: I didn't look into whether implementing my suggestion would be feasible -- just providing my two cents on what I think would be ideal.

Reply:
I think we could implement your suggestion by having an api for debug, and an api for prod. That would be problematic though because that choice would impact developers.

I agree that there is some risk that developers will interpret this more seriously than we want, and having to look up the error's documentation is an extra hurdle. Perhaps the best way forward is to just log the "error" to the console.

I'll make the change but am also interested in @ch40w31's thoughts.

Reply:
I think I've captured the logging in scenario with this snippet in `-[GIDAppCheck prepareForAppCheckWithCompletion:]`

```
    if ([self isPrepared]) {
#if DEBUG
      NSLog(@"`GIDAppCheck` is already prepared.");
#endif
```

`[self isPrepared]` tracks whether or not configure has been called and has succeeded. We only log during `DEBUG` builds.

Reply:
Per an offline conversation, I have [removed the `NSLog`](82b7ad8b4d6580965c3276a04dbf8e5adc6f3591) for previously in place for `DEBUG` builds.